### PR TITLE
Some-data-doesn't-adapt-on-Android

### PR DESCRIPTION
**Pull request by LendiHop**

### CODE REVIEW

File: src/components/activity-groups-list/activity-group-item/activity-group-dollar-amount-change/activity-group-dollar-amount-change.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        >
+          {inRange(amount, MAX_NEGATIVE_AMOUNT_VALUE, MIN_POSITIVE_AMOUNT_VALUE) && '≈ '}
+          {amount > 0 ? '+ ' : '- '}
+          {roundFiat(
+            new BigNumber(
+              amount > 0 ? clamp(amount, MIN_POSITIVE_AMOUNT_VALUE, Infinity) : clamp(amount, MAX_NEGATIVE_AMOUNT_VALUE)
+            ).abs()
+          ).toFixed()}
+          {' $'}
+        </Text>
```
Comment (unixvb) [inline src/components/activity-groups-list/activity-group-item/activity-group-dollar-amount-change/activity-group-dollar-amount-change.tsx:36]: could you reuse `AssetValueText` component instead?

File: src/components/activity-groups-list/activity-group-item/activity-group-item.tsx [hunk 2]
```diff
@@ -22,6 +42,52 @@ export const ActivityGroupItem: FC<Props> = ({ group }) => {
 
   const firstActivity = group[0] ?? emptyActivity;
 
+  const dispatch = useDispatch();
+  const getTokenMetadata = useTokenMetadataGetter();
+  const exchangeRates = useExchangeRatesSelector();
+
+  const nonZeroAmounts = useMemo(() => {
```
Comment (unixvb) [inline src/components/activity-groups-list/activity-group-item/activity-group-item.tsx:49]: create separate hook with this logic

File: src/components/activity-groups-list/activity-group-item/activity-group-item.tsx [hunk 3]
```diff
@@ -1,18 +1,38 @@
...
 import { ExternalLinkButton } from '../../icon/external-link-button/external-link-button';
 import { PublicKeyHashText } from '../../public-key-hash-text/public-key-hash-text';
 import { ActivityGroupAmountChange } from './activity-group-amount-change/activity-group-amount-change';
+import { ActivityGroupDollarAmountChange } from './activity-group-dollar-amount-change/activity-group-dollar-amount-change';
 import { useActivityGroupItemStyles } from './activity-group-item.styles';
 import { ActivityGroupType } from './activity-group-type/activity-group-type';
 import { ActivityStatusBadge } from './activity-status-badge/activity-status-badge';
 import { ActivityTime } from './activity-time/activity-time';
 
+export interface NonZeroAmounts {
```
Comment (unixvb) [inline src/components/activity-groups-list/activity-group-item/activity-group-item.tsx:26]: extract interface into seperate file

File: src/utils/number.util.ts [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
-  roundingMode: BigNumber.RoundingMode = BigNumber.ROUND_DOWN,
-  decimalPlace = 6
-) => {
+export const formatAssetAmount = (amount: BigNumber, decimalPlace = 6) => {
   if (isNaN(amount.toNumber())) {
     return '';
   }
 
-  return amount.decimalPlaces(amount.abs().lt(1000) ? decimalPlace : 2, roundingMode).toFixed();
+  return amount.decimalPlaces(amount.abs().lt(1000) ? decimalPlace : 2).toFixed();
```
Comment (unixvb) [inline src/utils/number.util.ts:9]: add `BigNumber.ROUND_DOWN` ad third argument

File: src/utils/big-number.utils.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { BigNumber } from 'bignumber.js';
+
+//Math.max and Math.min analogs
+const bigIntMax = (...args: BigNumber[]) => args.reduce((m, e) => (e > m ? e : m));
+const bigIntMin = (...args: BigNumber[]) => args.reduce((m, e) => (e < m ? e : m));
```
Comment (unixvb) [inline src/utils/big-number.utils.ts:5]: use `BigNumber` `isGreaterThan()` method

File: src/components/activity-groups-list/activity-group-item/activity-group-dollar-amount-change/activity-group-dollar-amount-change.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+export const ActivityGroupDollarAmountChange: FC<Props> = ({ nonZeroAmounts }) => {
+  const styles = useActivityGroupDollarAmountChangeStyles();
+
+  return (
+    <View style={styles.container}>
+      {nonZeroAmounts.dollarSums.map((amount, index) => (
+        <FormattedAmount
+          key={index}
+          amount={new BigNumber(amount)}
```
Comment (unixvb) [inline src/components/activity-groups-list/activity-group-item/activity-group-dollar-amount-change/activity-group-dollar-amount-change.tsx:22]: fix `TODO` and remove this BigNumber reation

**Review by andrewkryshtal — APPROVED** [review]

**Review by andrewkryshtal — APPROVED** [review]

File: src/utils/number.util.spec.ts [hunk 7]
```diff
@@ -21,24 +21,12 @@ describe('formatAssetAmount', () => {
...
-  });
-
-  it('should format positive bignumber more than 1000 value and return string with 2 decimals using rounding mode up', () => {
-    expect(formatAssetAmount(bigNumberMoreThanThousand, BigNumber.ROUND_UP)).toEqual('10 000.26');
-  });
-
-  it('should format positive bignumber less than 1000 value and return string with 1 decimal using rounding mode up', () => {
-    expect(formatAssetAmount(new BigNumber(100.255133), BigNumber.ROUND_UP, 1)).toEqual('100.3');
-  });
-
```
Comment (unixvb) [inline src/utils/number.util.spec.ts:35]: return tests

File: src/utils/big-number.utils.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { BigNumber } from 'bignumber.js';
+
+//Math.max and Math.min analogs
```
Comment (unixvb) [inline src/utils/big-number.utils.ts:3]: remove this comment

File: src/hooks/use-non-zero-amounts.hook.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        amounts.push({
+          parsedAmount,
+          isPositive,
+          symbol,
+          exchangeRate
+        });
+
+        if (amounts.length === 2 && amounts[0].isPositive) {
+          [amounts[0], amounts[1]] = [amounts[1], amounts[0]];
+        }
```
Comment (unixvb) [inline src/hooks/use-non-zero-amounts.hook.ts:54]: remove this ordering

**Review by unixvb — CHANGES_REQUESTED** [review]

**Review by unixvb — APPROVED** [review]

**Pull request merged**